### PR TITLE
AMPI: Clean up built-in attributes

### PR DESCRIPTION
**Pull request by evan-charmworks**
- Fix crash when getting builtin attributes from C/C++
- Disallow setting builtin attributes
- Separate Comm, Type, and Win builtin attribute checking
- Consider Win attributes to be per-window instead of per-PE
- Code organization cleanups

### CODE REVIEW

**Comment by evan-charmworks** [comment]
Fixes MPI_TAG_UB crash seen in C/C++ with mpi4py.

I tested all Comm attributes from both C++ and Fortran.

**Comment by matthiasdiener** [comment]
Related to #2823.

File: src/libs/ck-libs/ampi/ampi.C [hunk 1]
```diff
@@ -1722,11 +1722,11 @@ bool ampiParent::getBuiltinAttribute(int keyval, void *attribute_val) noexcept {
...
     if (keyval == MPI_WIN_BASE)
-      *((void**)attribute_val) = *win_base_storage;
+      *(void ***)attribute_val = win_base_storage;
     else if (keyval == MPI_WIN_SIZE)
-      *(MPI_Aint *)attribute_val = *win_size_storage;
+      *(MPI_Aint **)attribute_val = win_size_storage;
     else
-      *(int *)attribute_val = *kv_builtin_storage;
+      *(int **)attribute_val = kv_builtin_storage;
     return true;
```
Comment (matthiasdiener) [inline src/libs/ck-libs/ampi/ampi.C:1730]: This code would be quite a bit simpler if kv_get_builtin was merged into this function...

Comment (evan-charmworks) [reply src/libs/ck-libs/ampi/ampi.C]: Done

**Review by matthiasdiener — APPROVED** [review]

**Review by stwhite91 — APPROVED** [review]

**Pull request merged**